sdcm/utils/common.py

```diff
@@ -772,18 +774,19 @@ def clean_instances_aws(tags_dict, dry_run=False):
             name = tags.get("Name", "N/A")
             node_type = tags.get("NodeType")
             instance_id = instance['InstanceId']
-            argus_resources = [r for r in argus_run.run_info.resources.allocated_resources if r.name == name]
-            argus_resource = argus_resources[0] if len(argus_resources) > 0 else None
             if node_type and node_type == "sct-runner":
                 LOGGER.info("Skipping Sct Runner instance '%s'", instance_id)
                 continue
             LOGGER.info("Going to delete '{instance_id}' [name={name}] ".format(instance_id=instance_id, name=name))
             if not dry_run:
-                if argus_resource:
-                    argus_run.run_info.resources.detach_resource(argus_resource,
-                                                                 reason="clean-resources: Graceful Termination")
-                    argus_run.save()
                 response = client.terminate_instances(InstanceIds=[instance_id])
+                try:
+                    argus_client.terminate_resource(name=name, reason="clean-resources: Graceful Termination")
+                except ArgusClientError as exc:
+                    if len(exc.args) >= 3 and exc.args[2] == 404:
```

Comment:
Consider making ArgusClientError be more user-friendly. Using `exc.args >=3 and ex.args[2] == 404` is mysterious. Such things are unmaintainable.
e.g create new excetpion like "ArgusClientResourceDoesNotExistError" and throw it in client accordingly.

Reply:
Will do in 0.10.1